### PR TITLE
[Sage-1119] Form Section - Markup Refactor

### LINKED ISSUE #1119: [Improvement] Form Section Markup Refactor
## Description

As briefly mentioned in #674, it would be beneficial to update the Form Section component markup to make use of Sage Grid, GridCol, and Panel. There should be no visual changes with this refactor, but care should be taken to spot check uses in Kajabi Products to ensure there are no broken layouts.

### PR DESCRIPTION
**Pull request by anechol**
## Description

Refactors Form Section component to use Sage components for alignment.

## Screenshots

N/A

### Rails
- View [Rails component](http://localhost:4000/pages/component/form_section)
- Check that new markup displays as intended

### React
- View [Storybook](http://localhost:4100/?path=/docs/sage-formsection--default)
- Check that new markup displays as intended

## Testing in `kajabi-products`

(LOW) Refactors Form Section component to use Sage components for alignment. Visually, there should be no changes, but spot checks will be helpful.

## Related

Closes #1119

### CODE REVIEW

File: docs/lib/sage_rails/app/views/sage_components/_sage_form_section.html.erb [hunk 1]
```diff
@@ -1,30 +1,36 @@
...
-</div>
+    <%= sage_component SageGridCol, { breakpoint: :lg, size: "8" } do %>
+      <%= sage_component SagePanel, {} do %>
+        <div class="sage-form-section__content">
+          <%= component.content %>
+        </div>
+      <% end %>
+    <% end %>
+  <% end %>
+</div>
```
Comment (pixelflips) [inline docs/lib/sage_rails/app/views/sage_components/_sage_form_section.html.erb:36]: File is missing new line.

File: docs/app/views/examples/components/form_section/_preview.html.erb [hunk 2]
```diff
@@ -23,10 +23,10 @@
     title_tag: "h3"
   } do %>
     <%= sage_component SageAlert, {
-        title: "Form Section Content Will Go Here",
-        desc: "This component is designed to be very flexible and take any sort of content you want to pass to it.",
-        icon_name: "sage-icon-info-circle",
-      } %>
+      title: "Form Section Content Will Go Here",
+      desc: "This component is designed to be very flexible and take any sort of content you want to pass to it.",
+      icon_name: "sage-icon-info-circle",
```
Comment (pixelflips) [inline docs/app/views/examples/components/form_section/_preview.html.erb:28]: ❤️

**Review by pixelflips — APPROVED** [review]
LGTM! Nice improvement!

One small thing I noticed is that between around 800-991px the desc text bumps up against the `sage-form-section__content`. Not a show stopper, but it might be worth adding some spacing for those sizes (or maybe just add spacing to `sage-form-section__info` to prevent the need for media queries). 

Larger and smaller screens seem to not have the issue. Not a show stopper at all.

Screen Shot 2022-01-03 at 2 52 22 PM

**Review by monicawheeler — APPROVED** [review]
Looks good! I verified the responsive behavior Phillip saw at one point and also cannot duplicate the issue. :shipit:

**Comment by pixelflips** [comment]
@anechol I'd say we chalk up the spacing issue to a problem with my local. I checked again and also did not see it this time. Sorry for the confusion.

**Pull request merged**